app/src/androidTest/java/ch/epfl/sdp/appart/MapUITest.java

```diff
@@ -127,4 +131,68 @@ public void markerTest() throws InterruptedException {
                         " CLICKED")), 10000) | device.hasObject(By.descContains(card.getCity() + "CLICKED"));
         assertThat(isMarkerClicked, is(true));
     }
+
+    @Before
+    public void init() {
+        Intents.init();
+    }
+
+    @Test
+    public void markerClickTest() throws InterruptedException {
+
+
+        UiDevice device =
+                UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+
+        boolean foundMap = device.wait(Until.hasObject(By.desc("MAP READY")),
+                10000);
+        assertThat(foundMap, is(true));
+
+        Thread.sleep(2000);
```

Comment:
are these thread sleep still needed?

Reply:
Yes unfortunately. Some of them are even necessary in my local machine.